app/services/proofing/resolution/progressive_proofer.rb

```diff
@@ -0,0 +1,200 @@
+
+        attributes_aamva_can_pass = [:address, :dob, :state_id_number]
+        results_that_cannot_pass_aamva =
+          resolution_result.attributes_requiring_additional_verification - attributes_aamva_can_pass
+
+        results_that_cannot_pass_aamva.blank?
+      end
+
+      def threatmetrix_disabled_result
+        Proofing::DdpResult.new(
+          success: true,
+          client: 'tmx_disabled',
+          review_status: 'pass',
+        )
+      end
+
+      def out_of_aamva_jurisdiction_result
+        Proofing::StateIdResult.new(
+          errors: {},
+          exception: nil,
+          success: true,
+          vendor_name: 'UnsupportedJurisdiction',
+        )
+      end
+
+      def lexisnexis_ddp_proofer
+        @lexisnexis_ddp_proofer ||=
+          if IdentityConfig.store.lexisnexis_threatmetrix_mock_enabled
+            Proofing::Mock::DdpMockClient.new
+          else
+            Proofing::LexisNexis::Ddp::Proofer.new(
+              api_key: IdentityConfig.store.lexisnexis_threatmetrix_api_key,
+              org_id: IdentityConfig.store.lexisnexis_threatmetrix_org_id,
+              base_url: IdentityConfig.store.lexisnexis_threatmetrix_base_url,
+            )
+          end
+      end
+
+      def resolution_proofer
+        @resolution_proofer ||=
+          if IdentityConfig.store.proofer_mock_fallback
+            Proofing::Mock::ResolutionMockClient.new
+          else
+            Proofing::LexisNexis::InstantVerify::Proofer.new(
+              instant_verify_workflow: IdentityConfig.store.lexisnexis_instant_verify_workflow,
+              account_id: IdentityConfig.store.lexisnexis_account_id,
+              base_url: IdentityConfig.store.lexisnexis_base_url,
+              username: IdentityConfig.store.lexisnexis_username,
+              password: IdentityConfig.store.lexisnexis_password,
+              request_mode: IdentityConfig.store.lexisnexis_request_mode,
+            )
+          end
+      end
+
+      def state_id_proofer
+        @state_id_proofer ||=
+          if IdentityConfig.store.proofer_mock_fallback
+            Proofing::Mock::StateIdMockClient.new
+          else
+            Proofing::Aamva::Proofer.new(
+              auth_request_timeout: IdentityConfig.store.aamva_auth_request_timeout,
+              auth_url: IdentityConfig.store.aamva_auth_url,
+              cert_enabled: IdentityConfig.store.aamva_cert_enabled,
+              private_key: IdentityConfig.store.aamva_private_key,
+              public_key: IdentityConfig.store.aamva_public_key,
+              verification_request_timeout: IdentityConfig.store.aamva_verification_request_timeout,
+              verification_url: IdentityConfig.store.aamva_verification_url,
+            )
+          end
+      end
+
+      # Make a copy of pii with the user's state ID address overwriting the address keys
+      def with_state_id_address(pii)
+        pii&.transform_keys(SECONDARY_ID_ADDRESS_MAP)
```

Comment:
@NavaTim I'm not yet done responding to all your feedback, but I wanted to solicit your feedback on the nil-safe operator I just added to this line.  I was testing for a scenario where `pii` is nil,  which initially caused this line to throw a bug.  Thoughts on the nil-safe operator?

Reply:
We shouldn't support `nil` here, so I would say remove the `&` operator. It simplifies testing, and an error thrown in this location due to a `nil` value would signal a real bug in the application logic.

If this class is used correctly (and correct usage will be at least roughly demonstrated in the tests), then `proof` should always be called with a `Hash`, which will then be passed to `with_state_id_address`.

Reply:
If you do believe that support for `nil` is necessary, then that would also merit a change in the docblock for `proof` since `applicant_pii` only specifies `Hash` as a valid type right now. Side note - it looks like that docblock is on the class right now; it should be on the method instead.

Reply:
Hmmmmm.  If the progressive proofer shouldn't support scenarios where `applicant_pii` is nil, then why does `proof_with_threatmetrix_if_needed` include the line
```ruby
return threatmetrix_disabled_result unless applicant_pii
```

Reply:
Hi again!  I checked with @zachmargolis in slack, and there shouldn't be any scenarios where `applicant_pii` is nil.  As a result, I removed the nil check and the threatmetrix spec for nil `applicant_pii`.

Reply:
@eileen-nava Your last comment here is a little confusing since it looks like you added back the nil check. Based on your comment, I would expect the code to look more like the following.

```suggestion
        pii.transform_keys(SECONDARY_ID_ADDRESS_MAP)
```

Reply:
My mistake. 🤦🏻‍♀️ I needed to push the commit.